### DISCUSSION TITLE
[help] Uncaught (in promise) Error: Renderer.MissingLayout: with []

### DISCUSSION THREAD

**Discussion by cog25** [Myriad-Dreamin/typst.ts#619] [Q&A] [answered]
hello. I tried to use Typst.ts to compile and render Typst documents on the web, using example projects like Mitex. However, I'm getting the error.

```
Uncaught (in promise) Error: Renderer.MissingLayout: with []
    __wbg_new_70a2f23d1565c04c typst_ts_renderer.mjs:1767
    logError typst_ts_renderer.mjs:225
    __wbg_new_70a2f23d1565c04c typst_ts_renderer.mjs:1766
    svg_data typst_ts_renderer.mjs:1328
    renderSvg renderer.mts:919
    withinOptionSession renderer.mts:974
    renderSvg renderer.mts:901
    svg index.tsx:41
    runWithSession renderer.mts:1005
    App index.tsx:40
    untrack dev.js:518
    onMount dev.js:543
    runComputation dev.js:802
    updateComputation dev.js:780
    runTop dev.js:897
    runUserEffects dev.js:1011
    completeUpdates dev.js:966
    runUpdates dev.js:914
    completeUpdates dev.js:966
    runUpdates dev.js:915
    createRoot dev.js:207
    render dev.js:563
    <anonymous> index.tsx:56
typst_ts_renderer.mjs:1767:20
```

The repository that reproduced the error: https://github.com/cogarchive/typst-ts-why-not-working
Stackblitz is: https://stackblitz.com/~/cogarchive/typst-ts-why-not-working?file=src/index.tsx (open in new tab, then check console output)

**Comment by Myriad-Dreamin** [comment, marked as answer]
Hi, I explore it a bit and find it might be broken to set data and render at the same time. So you may render svg with code change:

```patch
+ kModule.manipulateData({
+     data: result,
+ });
return renderer.renderSvg({
-   artifactContent: result,
    format: "vector",
    renderSession: kModule,
})
```

Note you'll finally have to make such change because it only cache things in a single render session otherwise it will be very inefficient.

You may send an issue and I'll fix it later.

**Reply by cog25** [reply]
Your answer was really helpful. thank you.

(I also created an issue, but I'm not sure if I created it correctly.)

**Reply by Myriad-Dreamin** [reply]
Thank you. You did create it correctly.
